src/core/parser/csv_parser/csv_parser.cpp

```diff
@@ -110,12 +123,26 @@ std::string CSVParser::GetUnparsedLine(unsigned long long const line_index) {
 }
 
 std::vector<std::string> CSVParser::ParseString(std::string const& s) const {
+    std::string t;
+    for (size_t index = 0; index < s.size(); index++) {
+        if (s[index] == '\"' && index < s.size() - 1 && s[index + 1] == '\"' &&
+            (!t.size() || t[t.size() - 1] != '\\')) {
+            t.push_back('\\');
+        } else {
+            t.push_back(s[index]);
+        }
+    }
+
     std::vector<std::string> tokens;
     boost::escaped_list_separator<char> list_sep(escape_symbol_, separator_, quote_);
-    boost::tokenizer<boost::escaped_list_separator<char>> tokenizer(s, list_sep);
+    boost::tokenizer<boost::escaped_list_separator<char>> tokenizer(t, list_sep);
 
     for (auto& token : tokenizer) {
-        tokens.push_back(token);
+        if (token == "\"" && single_quote_is_empty_line) {
+            tokens.push_back("");
```

Comment:
```suggestion
            tokens.emplace_back();
```
This doesn't create and move a `std::string`